### PR TITLE
Report the fields we need

### PR DESCRIPTION
**Pull request by jhuebner-work**
This reports all the fields we need but needs to be more flexible and throw errors for bad values.
@wondersd

### CODE REVIEW

File: checkipa [hunk 1]
```diff
@@ -233,10 +237,24 @@ class IntegrityCheck(object):
...
+        if 'CFBundleShortVersionString' in self.data:
+            txt.append('  Release Version (CFBundleShortVersionString): %s' % (self.data['CFBundleShortVersionString']))
         if 'CFBundleVersion' in self.data:
-            txt.append('  Version: %s' % (self.data['CFBundleVersion']))
-
+            txt.append('  Build Version (CFBundleVersion): %s' % (self.data['CFBundleVersion']))
+        if ('Name' in self.profile):
+            txt.append('  Profile Name: %s' % (self.profile['Name']))
+        if ('application-identifier' in self.profile['Entitlements']):
+            appId = self.profile['Entitlements']['application-identifier'];
```
Comment (jhuebner-work) [inline checkipa:289]: This should be pulled out into a config file. and possibly show both the name and number.

File: checkipa [hunk 2]
```diff
@@ -255,7 +275,10 @@ class IntegrityCheck(object):
                 txt.append('  %s' % (udid))
         else:
             txt.append('  Provisioned Devices (0):')
-
+        txt.append('  Package size: %2.2f GB' % (self.ipa_size/1000000000.0))
```
Comment (jhuebner-work) [inline checkipa]: need constants for GB and MB

File: checkipa [hunk 3]
```diff
@@ -461,7 +488,28 @@ class IntegrityCheck(object):
         elif 'ProvisionedDevices' in self.profile:
             self.ipa_info['adhoc'] = True
         else:
-            self.errors['distmethod'] = True
+            self.ipa_info['app_store'] = True
+            #self.errors['distmethod'] = True
+    def check_ipa_file_size(self):
+        if self.ipa_size > 4000000000:
```
Comment (jhuebner-work) [inline checkipa]: need configurable constant for 4GB limit.

File: checkipa [hunk 4]
```diff
@@ -461,7 +488,28 @@ class IntegrityCheck(object):
...
+    def check_executable_size(self):
+        size_rx = re.compile(r'^\s+size\s+(\d+)',re.MULTILINE)
+        sizes = size_rx.findall(self.executable_info)
+        maxsize = -1
+        for size_str in sizes:
+            size = int(size_str)
+            if size > maxsize:
+                maxsize = size 
+        self.executable_size = int(maxsize)
+        if 7 >= 7 :
```
Comment (jhuebner-work) [inline checkipa:563]: need configurable constants for limits here, and need actual os version check

File: checkipa [hunk 5]
```diff
@@ -461,7 +488,28 @@ class IntegrityCheck(object):
...
+            if self.executable_size >= 60000000:
+                self.errors['size_exec_61'] = True
```
Comment (jhuebner-work) [inline checkipa:565]: need to actually handle these errors

File: checkipa [hunk 6]
```diff
@@ -605,6 +658,37 @@ class ParseIPA(object):
...
+        match = ParseIPA.plist_executable_rx.search(filenames[1])
+        base = match.group(1);
+        filename = 'Payload/%s.app/%s' % (base,base);
+        return filename
+    # end get_filename_from_ipa()
+
+    def get_executable_info(self):
+        with zipfile.ZipFile(self.ipa_filename) as ipa_zip:
+            exec_path=self.get_exec_filename_from_ipa()
+            with open('Exe', 'wb') as f:
```
Comment (jhuebner-work) [inline checkipa]: Need to dynamically generate and clean up filename instead of using Exe

File: checkipa [hunk 7]
```diff
@@ -678,7 +782,9 @@ def get_options():
     optp.add_option('-u', '--udids', action='store', dest='udids',
                     help='''check if udids are provisioned
                         ''')
-
+    optp.add_option('-t', '--teams', action='store', dest='teams',
```
Comment (jhuebner-work) [inline checkipa:933]: need to use this for application-identifier above

File: checkipa [hunk 8]
```diff
@@ -245,8 +263,10 @@ class IntegrityCheck(object):
         elif self.ipa_info['enterprise'] and not self.ipa_info['adhoc']:
```
Comment (djlaplante) [inline checkipa]: this should not be "elif", it should just be "if". An enterprise profile can have an expiration date

File: checkipa [hunk 9]
```diff
@@ -245,8 +263,10 @@ class IntegrityCheck(object):
...
             ipa_type = 'Enterprise'
         else:
-            ipa_type = 'unknown distribution type'
-            report_errors.append({'label': 'Distribution', 'description': 'unknown distribution type'})
+            if((len(provisioned_devices) == 0) and not(self.profile['Entitlements']['get-task-allow'])): 
+                ipa_type = 'AppStore Submission'
+            else:
+                report_errors.append({'label': 'Distribution', 'description': 'unknown distribution type'})
```
Comment (djlaplante) [inline checkipa:310]: within this block, ipa_type must be given a value (such as "unknown distribution type")

File: checkipa (hunk 2 above)
Comment (djlaplante) [inline checkipa]: self.ipa_size is not checked for validity before use

File: checkipa [hunk 10]
```diff
@@ -255,7 +275,10 @@ class IntegrityCheck(object):
                 txt.append('  %s' % (udid))
         else:
             txt.append('  Provisioned Devices (0):')
-
+        txt.append('  Package size: %2.2f GB' % (self.ipa_size/1000000000.0))
+        txt.append('  Executable size: %2.2f MB' % (self.executable_size/1000000.0))
+        txt.append('  MD5: %s ' % self.ipa_md5)
+
```
Comment (djlaplante) [inline checkipa:326]: The above 3 lines (278, 279, and 280) do not check validity of the variables self.ipa_size, self.executable_size, and self.ipa_md5 before they are used

File: checkipa [hunk 11]
```diff
@@ -625,6 +709,26 @@ class ParseIPA(object):
...
+            if provision_filename == '':
+                self.errors.append('Executable file not found in IPA')
+            else:
+                content = zip_obj.read(provision_filename)
+                match = ParseIPA.provision_xml_rx.search(content)
+                if (match is not None):
+                    provision_xml_content = match.group()
+                    data = plistlib.readPlistFromString(provision_xml_content)
+                else:
+                    self.errors.append('unable to parse embedded.mobileprovision file')
```
Comment (djlaplante) [inline checkipa:875]: data should be set in this block, even if its an error message

File: checkipa (hunk 2 above)
Comment (wondersd) [inline checkipa]: File sizes/prefixes are all powers of 2 not powers of 10. Therefor to convert to GB from what is assumed to be bytes is dividing by 1073741824 or 2^30.

Same goes for the 4GB check later on 4GB is 4294967296 bytes (4*2^30) not 4000000000 bytes. It works in that it wont allow an ipa above the known limit, but is further limiting them by almost a third of a GB.

**Comment by jhuebner-work** [comment]
I think we should merge this time

File: checkipa (hunk 2 above)
Comment (jhuebner-work) [inline checkipa]: @wondersd, I think you mean GibiByte:
https://en.wikipedia.org/wiki/Gibibyte 
Apple spec says GB not GiB, and given the ambiguity, I would rather be more restrictive.

File: checkipa (hunk 9 above)
Comment (jhuebner-work) [inline checkipa:310]: done

File: checkipa (hunk 10 above)
Comment (jhuebner-work) [inline checkipa:326]: added if  statements

File: checkipa (hunk 11 above)
Comment (jhuebner-work) [inline checkipa:875]: done

**Comment by jhuebner-work** [comment]
obsolete and redone in another branch

**Comment by nick-sophinos** [comment]
Hi John I am at an eye doctor appt. and will be a the office by 12:00 unless the dilated pupils keep me from seeing well. Never had one before so I am not sure.  I will let you know when the appt is over.

Thanks,

Nick

**Pull request closed without merging**